Core/Inc/Communications/Packets/Packet_generation/DataTemplate.hpp

```diff
@@ -1,21 +1,26 @@
 #include "ST-LIB.hpp"
 
-//Data packets for %board%
-class DataPacket{
-    
-%enums%
+//Order packets for {{board}}
+class DataPackets{
+{% for enum in enums %}
+{{enum.enum}}
+{% endfor %}
 
     private:
-        constexpr static size_t size =%size%;
+        constexpr static size_t size ={{size}};
         uint32_t id{0};
     public:
         std::array<StackPacket*,size> packets; 
-%packetnames%
+        {%for packet in packets%}StackPacket* {{packet.name}};
+        {% endfor %}
         
-    DataPacket(%data%)
+    DataPackets({{data}})
 {
 
-%packets%
+{% for packet in packets %}{{packet.name}} = new StackPacket({{packet.id}}{% if packet.data%},{{packet.data}}{% endif%});
+packets[id] = {{packet.name}};
+id++;
+{% endfor %}
```

Comment:
```suggestion
{% for i, packet in enumerate(packets) %}
{{packet.name}} = new StackPacket(
    {{packet.id}}{% for values in packet.data %},
    {{packet.data}}{% endfor %}
);
packets[i] = {{packet.name}};
{% endfor %}
```
Again I don't know the syntax specifics, but this is much cleaner if its possible

Reply:
also notice you don't need the id variable now, so one thing less you don't need